modules/gdprEnforcement.js

```diff
@@ -11,39 +11,72 @@ import includes from 'core-js-pure/features/array/includes.js';
+import events from '../src/events.js';
+import { EVENTS } from '../src/constants.json';
 
-const purpose1 = 'storage';
+const TCF2 = {
+  'purpose1': { id: 1, name: 'storage' },
+  'purpose2': { id: 2, name: 'basicAds' }
+}
+
+const DEFAULT_RULES = [{
+  purpose: 'storage',
+  enforcePurpose: true,
+  enforceVendor: true,
+  vendorExceptions: []
+}, {
+  purpose: 'basicAds',
+  enforcePurpose: true,
+  enforceVendor: true,
+  vendorExceptions: []
+}];
 
+let purpose1Rule;
+let purpose2Rule;
 let addedDeviceAccessHook = false;
 let enforcementRules;
 
-function getGvlid() {
+function getGvlid(bidderCode) {
   let gvlid;
-  const bidderCode = config.getCurrentBidder();
+  bidderCode = bidderCode || config.getCurrentBidder();
   if (bidderCode) {
     const bidder = adapterManager.getBidAdapter(bidderCode);
-    gvlid = bidder.getSpec().gvlid;
+    if (bidder && bidder.getSpec) {
+      gvlid = bidder.getSpec().gvlid;
+    }
   } else {
     utils.logWarn('Current module not found');
   }
   return gvlid;
 }
 
 /**
- * This function takes in rules and consentData as input and validates against the consentData provided. If it returns true Prebid will allow the next call else it will log a warning
- * @param {Object} rules enforcement rules set in config
- * @param {Object} consentData gdpr consent data
+ * This function takes in a rule and consentData and validates against the consentData provided. If it returns true Prebid will allow the next call else it will log a warning
+ * @param {Object} rule - enforcement rules set in config
+ * @param {Object} consentData - gdpr consent data
+ * @param {string=} currentModule - Bidder code of the current module
+ * @param {number=} gvlId - GVL ID for the module
  * @returns {boolean}
  */
-function validateRules(rule, consentData, currentModule, gvlid) {
-  // if vendor has exception => always true
+export function validateRules(rule, consentData, currentModule, gvlId) {
+  const purposeId = TCF2[Object.keys(TCF2).filter(purposeName => TCF2[purposeName].name === rule.purpose)[0]].id;
+
+  // return 'true' if vendor present in 'vendorExceptions'
   if (includes(rule.vendorExceptions || [], currentModule)) {
     return true;
   }
-  // if enforcePurpose is false or purpose was granted isAllowed is true, otherwise false
-  const purposeAllowed = rule.enforcePurpose === false || utils.deepAccess(consentData, 'vendorData.purpose.consents.1') === true;
-  // if enforceVendor is false or vendor was granted isAllowed is true, otherwise false
-  const vendorAllowed = rule.enforceVendor === false || utils.deepAccess(consentData, `vendorData.vendor.consents.${gvlid}`) === true;
+
+  const purposeConsent = utils.deepAccess(consentData, `vendorData.purpose.consents.${purposeId}`);
+  const vendorConsent = utils.deepAccess(consentData, `vendorData.vendor.consents.${gvlId}`);
+  const liTransparency = utils.deepAccess(consentData, `vendorData.purpose.legitimateInterests.${purposeId}`);
+
```

Comment:
Comment: "Since vendor exceptions have already been handled, the purpose as a whole is allowed if it's not being enforced or the user has consented. Similar with vendors."